src/common/utils.py

```diff
@@ -301,18 +302,26 @@ def _fetch_pricing_file(region, proxy_config):
         raise
 
 
-def _get_vcpus_by_instance_type(instances, instance_type):
+def _get_vcpus_mem_by_instance_type(instances, instance_type):
     """
-    Get vcpus for the given instance type from the pricing file.
+    Get vcpus and memory for the given instance type from the pricing file.
 
     :param instances: dictionary conatining the content of the instances file
     :param instance_type: The instance type to search for
-    :return: the number of vcpus for the given instance type
+    :return: the number of vcpus for the given instance type, memory(in MiB) for the instance type
     :raise CriticalError if unable to find the given instance or whatever error.
     """
     try:
         vcpus = int(instances[instance_type]["vcpus"])
-        return vcpus
+        log.info("Instance {0} has {1} vcpus.".format(instance_type, vcpus))
+        # Rounding down on memory in each conversion step to avoid "low real_memory" error in slurm
+        memory = instances[instance_type]["memory"]
```

Comment:
what about printing an error and skipping memory config if not present in the file?